src/core/lint-hrefs-ids.js

```diff
@@ -0,0 +1,26 @@
+// Module core/lint-hrefs-ids
+// Warns about href's that do not have corresponding id's
+
+import { pub } from "core/pubsubhub";
+
+export const name = "core/lint-hrefs-ids";
+
+export function run(conf, doc, cb) {
+  Array.from(doc.querySelectorAll('a[href]'))
+    .map(a => ({ a, href: a.getAttribute('href') }))
+    .filter(({ href }) => Boolean(href))
```

Comment:
The intent here is really to check if this is a local reference, so we should do that properly (below).  I also realized that the current code won't handle cases where the `a.href` is to the same document, but set using an absolute URL, like this:

```HTML
<a href="http://example.com/#foo">this is not handled at the moment</a>
```

Thus, we really want:

```JS
Array.from(document.querySelectorAll("a[href]"))
  .filter(
    ({ pathname, origin, hash }) =>
      hash && // also filters out just "#" 🎉
      origin === doc.location.origin &&
      doc.location.pathname === pathname
  )
  .filter(({ hash }) => !doc.getElementById(decodeURIComponent(hash.slice(1))))
  .forEach(a => {
     // The rest
   });
```